test/widget_tests/utils/popup_manager/popup_manager_test.dart

```diff
@@ -0,0 +1,631 @@
+import 'dart:async';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_core/flutter_core.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('Popup Manager Tests', () {
+    testWidgets(
+      'Loader Test',
+      (tester) async {
+        final navigatorKey = GlobalKey<NavigatorState>();
+        final popupManager = PopupManager(navigatorKey: navigatorKey);
+        await tester.pumpWidget(
+          MaterialApp(
+            navigatorKey: navigatorKey,
+            home: const Scaffold(
+              body: Center(),
+            ),
+          ),
+        );
+
+        popupManager.showLoader();
+        await tester.pump();
+        expect(find.byType(CircularProgressIndicator), findsOneWidget);
+        popupManager.hidePopup<void>();
+        await tester.pump();
+        expect(find.byType(CircularProgressIndicator), findsNothing);
+        final id = UniqueKey().toString();
+        popupManager.showLoader(id: id);
+        await tester.pump();
+        expect(find.byType(CircularProgressIndicator), findsOneWidget);
+        popupManager.hidePopup<void>(id: id);
+        await tester.pump();
+        expect(find.byType(CircularProgressIndicator), findsNothing);
+      },
+    );
+
+    testWidgets(
+      'ModalBottomSheet Test',
```

Comment:
Nit: For instance, if we're testing adaptiveDialog, defining it as a group and organizing related tests under it improves code readability. Additionally, naming becomes more understandable.

Example: 
```dart
group('Loader Tests', () {
    testWidgets('Loader Test',(tester) async {});
    testWidgets('Loader Test with ID',(tester) async {});
  });
 ```